pkg/clustermanager/cluster_manager.go

```diff
@@ -209,12 +210,19 @@ func (c *ClusterManager) CreateWorkloadCluster(ctx context.Context, managementCl
 		// the condition external etcd ready if true indicates that all etcd machines are ready and the etcd cluster is ready to accept requests
 	}
 
-	logger.V(3).Info("Waiting for control plane to be ready")
-	err = c.clusterClient.WaitForControlPlaneReady(ctx, managementCluster, ctrlPlaneWaitStr, workloadCluster.Name)
+	logger.V(3).Info("Waiting for workload kubeconfig secret to be ready", "cluster", workloadCluster.Name)
+	err = c.Retrier.Retry(
+		func() error {
+			err = c.clusterClient.KubeconfigSecretAvailable(ctx, managementCluster.KubeconfigFile, workloadCluster.Name, constants.EksaSystemNamespace)
+			return err
+		},
+	)
+
 	if err != nil {
-		return nil, fmt.Errorf("error waiting for workload cluster control plane to be ready: %v", err)
+		return nil, fmt.Errorf("error checking availability of kubeconfig secret: %v", err)
 	}
 
+	logger.V(3).Info("Waiting for workload kubeconfig generation", "cluster", workloadCluster.Name)
 	err = c.Retrier.Retry(
```

Comment:
Do we need this retry around `generateWorkloadKubeconfig`? If we are already waiting for the kubeconfig to be available, that should be enough, right?